### PR TITLE
Static analysis fixes

### PR DESCRIPTION
**Pull request by RensR**
This PR addresses various static analysis findings, mostly to reduce gas and contract bytecode size. Both before and after use 3600 optimizations. Due to the saves space in the onRamp we can bump the number of optimizations.

Each commit addresses a different change, and each commit contains the gas snapshot to illustrate the impact of that particular change.

Before 

```
| EVM2EVMMultiOffRamp             | 24.665    | -0.089      |
| EVM2EVMMultiOnRamp              | 20.885    | 3.691       |
| EVM2EVMOffRamp                  | 23.092    | 1.484       |
| EVM2EVMOnRamp                   | 24.432    | 0.144       |

2E:test_E2E_3MessagesSuccess_gas() (gas: 1.104.821)
MultiRampsE2E:test_E2E_3MessagesSuccess_gas() (gas: 1.424.394)
```

After 

```
| EVM2EVMMultiOffRamp             | 24.538    | 0.038       |
| EVM2EVMMultiOnRamp              | 20.841    | 3.735       |
| EVM2EVMOffRamp                  | 22.968    | 1.608       |
| EVM2EVMOnRamp                   | 24.351    | 0.225       |

E2E:test_E2E_3MessagesSuccess_gas() (gas: 1.102.679)
MultiRampsE2E:test_E2E_3MessagesSuccess_gas() (gas: 1.422.669)
```

note: comment slightly outdated since the last merge. Overall diff should still be the same

### CODE REVIEW

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol [hunk 1]
```diff
@@ -818,8 +832,8 @@ contract EVM2EVMMultiOffRamp is ITypeAndVersion, MultiOCR3Base {
     // We call the pool with exact gas to increase resistance against malicious tokens or token pools.
     // We protects against return data bombs by capping the return data size at MAX_RET_BYTES.
     (bool success, bytes memory returnData,) = CallWithExactGas._callWithExactGasSafeReturnData(
-      abi.encodeWithSelector(
-        IPoolV1.releaseOrMint.selector,
+      abi.encodeCall(
```
Comment (RensR) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:838]: Saves gas AND adds type safety, no brainer

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMOffRamp.sol [hunk 2]
```diff
@@ -38,7 +38,7 @@ contract EVM2EVMOffRamp is IAny2EVMOffRamp, AggregateRateLimiter, ITypeAndVersio
   error AlreadyExecuted(uint64 sequenceNumber);
   error ZeroAddressNotAllowed();
   error CommitStoreAlreadyInUse();
-  error ExecutionError(bytes error);
+  error ExecutionError(bytes err);
```
Comment (RensR) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMOffRamp.sol:41]: To remove shadowing the `error` type

File: contracts/src/v0.8/ccip/test/helpers/MessageHasher.sol [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {Internal} from "../../libraries/Internal.sol";
 
 // MessageHasher is a contract that provides a function to hash an EVM2EVMMessage.
 contract MessageHasher {
-  function hash(Internal.EVM2EVMMessage memory msg, bytes32 metadataHash) public pure returns (bytes32) {
-    return Internal._hash(msg, metadataHash);
+  function hash(Internal.EVM2EVMMessage memory message, bytes32 metadataHash) public pure returns (bytes32) {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/helpers/MessageHasher.sol]: remove shadowing `msg`

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol [hunk 4]
```diff
@@ -283,8 +283,10 @@ contract EVM2EVMMultiOffRamp is ITypeAndVersion, MultiOCR3Base {
       for (uint256 msgIndex = 0; msgIndex < numMsgs; ++msgIndex) {
         uint256 newLimit = msgGasLimitOverrides[msgIndex];
         // Checks to ensure message cannot be executed with less gas than specified.
-        if (newLimit != 0 && newLimit < report.messages[msgIndex].gasLimit) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:287]: q: do these changes actually help gas / contract size significantly? Seems less readable this way

Comment (RensR) [reply contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:287]: Definitely one of the more questionable changes, I agree. Minor gas savings, no size savings

Comment (RensR) [reply contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:287]: Some make sense, some less so. If it's a special case (e.g. nonces, gas overrides) I think != 0 makes it *more* readable. If it's a length, I changed it back to >0

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol [hunk 5]
```diff
@@ -438,19 +444,23 @@ contract EVM2EVMMultiOffRamp is ITypeAndVersion, MultiOCR3Base {
...
-          && originalState != Internal.MessageExecutionState.UNTOUCHED
-      ) {
-        // If manual execution fails, we revert the entire transaction, unless the originalState is UNTOUCHED as we
-        // would still be making progress by changing the state from UNTOUCHED to FAILURE.
-        revert ExecutionError(message.header.messageId, returnData);
+      if (manualExecution) {
+        if (
+          newState == Internal.MessageExecutionState.FAILURE
+            && originalState != Internal.MessageExecutionState.UNTOUCHED
+        ) {
```
Comment (RayXpub) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol]: q: should we also split this if statement into 2 separate ones?

**Review by RayXpub — APPROVED** [review]

**Review by 0xsuryansh — APPROVED** [review]

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol [hunk 6]
```diff
@@ -284,8 +284,10 @@ contract EVM2EVMMultiOffRamp is ITypeAndVersion, MultiOCR3Base {
       for (uint256 msgIndex = 0; msgIndex < numMsgs; ++msgIndex) {
         uint256 newLimit = msgGasLimitOverrides[msgIndex];
         // Checks to ensure message cannot be executed with less gas than specified.
-        if (newLimit != 0 && newLimit < report.messages[msgIndex].gasLimit) {
-          revert InvalidManualExecutionGasLimit(report.sourceChainSelector, msgIndex, newLimit);
+        if (newLimit != 0) {
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:287]: q: are the gas / contract size changes significant for splitting if statements? I find `a && b && c` much more readable in most cases

**Review by elatoskinas — APPROVED** [review]

**Pull request merged**